frontend/components/ui/SmartLink.tsx

```diff
@@ -0,0 +1,39 @@
+import { Box, BoxProps as ChakraBoxProps, forwardRef } from '@chakra-ui/react';
+import NextLink, { LinkProps } from 'next/link';
+
+type SmartLinkProps = Pick<LinkProps, 'legacyBehavior'> &
+   ChakraBoxProps & {
+      href: string;
+      behaviour?: LinkBehavior;
+   };
+
+type LinkBehavior = 'auto' | 'reload' | 'fast';
```

Comment:
What does 'fast' do?

Reply:
It's supposed to represent the normal Next.js behaviour, it should behave like just using a Next.js link, although practically speaking this component should probably be used only with link behaviour `auto`.